### PR TITLE
Make the syntax of example commands more consistent between the READMEs.

### PR DESCRIPTION
**Pull request by dlyongemallo**
# Description

The various READMEs use slightly different (but equally valid) syntax for example conan and gcovr commands. This makes it a bit more difficult for a user to see the relationships between them, especially when looking at their command history after having followed the steps in the READMEs. This PR unifies the syntax to the predominant one (`--param=value` instead of `--param value`, and quotes around options which is required by zsh but optional in other shells).

# Related issues

N/A

# Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented hard-to-understand parts of my code.
- [x] I have made corresponding changes to the public API documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have updated the changelog with any user-facing changes.

### CODE REVIEW

File: tket/README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ Assuming the prerequisite tools and configuration described in the top-level
 README, to build `tket` (from the root directory of the repo):
 
 ```shell
-conan create tket --user=tket --channel=stable --build=missing -o "boost/*":header_only=True
+conan create tket --user=tket --channel=stable --build=missing -o boost/*:header_only=True
```
Comment (cqc-alec) [inline tket/README.md:13]: I recall that there was a reason for these quotes around `boost/*`, that they were necessary on MacOS systems running zsh.

File: tket/README.md [hunk 2]
```diff
@@ -38,7 +38,7 @@ wget http://mirrors.ctan.org/graphics/pgf/contrib/quantikz/tikzlibraryquantikz.c
 To build tket and its dependencies and run all unit and property tests, use:
 
 ```shell
-conan build tket --build=missing -o "boost/*":header_only=True -o with_all_tests=True
+conan build tket --build=missing -o boost/*:header_only=True -o with_all_tests=True
```
Comment (cqc-alec) [inline tket/README.md:41]: See above.

File: tket/README.md [hunk 3]
```diff
@@ -87,8 +87,8 @@ If using gcc and with gcovr installed (`pip install gcovr`), a test coverage
 report can be generated using the following sequence of commands:
 
 ```shell
-conan build tket --user=tket --channel=stable -s build_type=Debug --build=missing -o "boost/*":header_only=True -o "tket/*":profile_coverage=True -o "test-tket/*":with_coverage=True -o with_test=True -of build/tket
+conan build tket --user=tket --channel=stable -s build_type=Debug --build=missing -o boost/*:header_only=True -o tket/*:profile_coverage=True -o test-tket/*:with_coverage=True -o with_test=True -of build/tket
```
Comment (cqc-alec) [inline tket/README.md:90]: See above (for quotes around arguments containing asterisks).

**Review by cqc-alec — CHANGES_REQUESTED** [review]
Thanks! I think we do need to keep the quotes around arguments containing asterisks, which are treated differently by zsh.

File: tket/README.md (hunk 3 above)
Comment (dlyongemallo) [reply tket/README.md]: Done.

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**